src/app/components/chat/room-channel.ts

```diff
@@ -262,22 +267,35 @@ export async function createChatRoomChannel(
 			return;
 		}
 
-		const edited = new ChatMessage(data);
-
-		const index = client.messages[roomId].findIndex(msg => msg.id === data.id);
-		const message = client.messages[roomId][index];
+		const message = room.value.messages.find(i => i.id === data.id);
+		if (!message) {
+			return;
+		}
 
+		const edited = new ChatMessage(data);
 		message.content = edited.content;
 		message.edited_on = edited.edited_on;
 	}
 
-	function _onMemberLeave(data: MemberLeavePayload) {
-		const roomMembers = client.roomMembers[roomId];
+	function _onMemberInc(data: MemberIncPayload) {
+		room.value.member_count += data.amount;
+	}
+
+	function _onMemberDec(data: MemberDecPayload) {
+		room.value.member_count -= data.amount;
+	}
```

Comment:
Seems to be some issues with syncing these properly. Testing over a couple tabs, different users may have different counts (4 members counted instead of 2), and refreshing one of the tabs can bring a member count from 2 to 0.